internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Analyze.scala

```diff
@@ -50,8 +50,16 @@ private[sbt] object Analyze {
          binaryClassName = classFile.className;
          loadedClass <- load(binaryClassName, Some("Error reading API from class file"))) {
       binaryClassNameToLoadedClass.update(binaryClassName, loadedClass)
-      val srcClassName = binaryToSourceName(loadedClass)
-        .orElse(Option(loadedClass.getEnclosingClass).flatMap(binaryToSourceName))
+
+      def loadEnclosingClass(clazz: Class[_]): Option[String] = {
+        binaryToSourceName(clazz) match {
+          case None if loadedClass.getEnclosingClass != null =>
```

Comment:
I think you mean `clazz.getEnclosingClass != null` here.

Reply:
How was the test case even passing with this error? This would mean that more than one level of nesting would not be handled.

Reply:
Tests passes since if `loadedClass` is nested class (regardless how deep) it will be always null.

The only problem with this is that in case that for some reason we don't have source mapping for top enclosing class we will get NullPointerException here. But anyway this was stupid mistake.